example-mlem-get-started/generate.sh

```diff
@@ -0,0 +1,156 @@
+#!/bin/bash
+# See https://dvc.org/doc/tutorials/get-started
+
+# Setup script env:
+#   e   Exit immediately if a command exits with a non-zero exit status.
+#   u   Treat unset variables as an error when substituting.
+#   x   Print commands and their arguments as they are executed.
+set -eux
+
+HERE="$( cd "$(dirname "$0")" ; pwd -P )"
+REPO_NAME="example-mlem-get-started"
+
+BUILD_PATH="$HERE/build"
+
+pushd $BUILD_PATH
+if [ ! -d "$BUILD_PATH/.venv" ]; then
+  exit 1
+  virtualenv -p python3 .venv
+  export VIRTUAL_ENV_DISABLE_PROMPT=true
+  source .venv/bin/activate
+  echo '.venv/' > .gitignore
+  pip install gitpython
+  pip install "git+https://github.com/iterative/mlem#egg=mlem[all]" --use-deprecated=legacy-resolver
+  pip install -r $HERE/code/src/requirements.txt
+fi
+popd
+
+source $BUILD_PATH/.venv/bin/activate
+
+REPO_PATH="$HERE/build/$REPO_NAME"
+
+if [ -d "$REPO_PATH" ]; then
+  echo "Repo $REPO_PATH already exists, please remove it first."
+  exit 1
+fi
+
+TOTAL_TAGS=15
+STEP_TIME=100000
+BEGIN_TIME=$(( $(date +%s) - ( ${TOTAL_TAGS} * ${STEP_TIME}) ))
+export TAG_TIME=${BEGIN_TIME}
+export GIT_AUTHOR_DATE=${TAG_TIME}
+export GIT_COMMITTER_DATE=${TAG_TIME}
+tick(){
+  export TAG_TIME=$(( ${TAG_TIME} + ${STEP_TIME} ))
+  export GIT_AUTHOR_DATE=${TAG_TIME}
+  export GIT_COMMITTER_DATE=${TAG_TIME}
+}
+
+export GIT_AUTHOR_NAME="Mikhail Sveshnikov"
+export GIT_AUTHOR_EMAIL="mike0sv@gmail.com"
+export GIT_COMMITTER_NAME="$GIT_AUTHOR_NAME"
+export GIT_COMMITTER_EMAIL="$GIT_AUTHOR_EMAIL"
+
+
+
+mkdir -p $REPO_PATH
+pushd $REPO_PATH
+
+git init -b main
+cp $HERE/code/README.md .
+cp $HERE/.gitattributes .
+git add .
+tick
+git commit -m "Initialize Git repository"
+git tag -a "0-git-init" -m "Git initialized."
+
+mlem init
+tick
+git add .mlem/config.yaml
+git commit -m "Initialize MLEM project"
+git tag -a "1-mlem-init" -m "MLEM initialized."
+
+
+cp $HERE/code/src/requirements.txt .
+cp $HERE/code/src/prepare.py .
+git add .
+python prepare.py
+git add .mlem/dataset
+tick
+git commit -m "Create data stage"
```

Comment:
in this case it can be as simple as "Add data" ? (no need to use "stage")

Reply:
ah, I see, it actually prepares data ... seems a bit too complicated ... should we just pre-bundle those files with code?

Reply:
we want to show how to save data also. So reading a file, and then re-saving it to mlem is even more compicated